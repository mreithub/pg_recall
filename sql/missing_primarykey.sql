--
-- Tries to enable pg_recall on a table without primary key (which should trigger an exception)
--

BEGIN;

CREATE TABLE config (
	key VARCHAR(100) NOT NULL,
	value TEXT NOT NULL
);

SELECT recall_enable('config', null);

COMMIT;
