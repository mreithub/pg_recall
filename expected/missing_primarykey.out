--
-- Tries to enable pg_recall on a table without primary key (which should trigger an exception)
--
BEGIN;
CREATE EXTENSION recall;
CREATE TABLE config (
	key VARCHAR(100) NOT NULL,
	value TEXT NOT NULL
);
SELECT recall_enable('config', null);
ERROR:  You need a primary key on your table if you want to use pg_recall (table: config)!
COMMIT;
